POEM_081.md

```diff
@@ -100,124 +206,40 @@ class SubproblemComp(om.ExplicitComponent):
+
+    def setup(self):        
+        p = self._subprob = om.Problem(**self.prob_args)
+        p.model.add_subsystem('subsys', self.model, promotes=['*'])
+        
+        p.setup(force_alloc_complex=self._problem_meta['force_alloc_complex'])
         p.final_setup()
 
-        model_inputs = p.model.list_inputs(out_stream=None, prom_name=True,
-                                           units=True, shape=True, desc=True)
-        model_outputs = p.model.list_outputs(out_stream=None, prom_name=True,
-                                             units=True, shape=True, desc=True)
-
-        # store model inputs/outputs as dictionary with keys as the promoted name
-        model_inputs = {meta['prom_name']: meta for _, meta in model_inputs}
-        model_outputs = {meta['prom_name']: meta for _, meta in model_outputs}
-
-        # loop through inputs and make sure they're valid for use
-        for inp in inputs:
-            if isinstance(inp, tuple):
-                # check if variable already exists in self.options['inputs']
-                # i.e. no repeated input variable names
-                if inp[1] in self.options['inputs']:
-                    raise Exception(f'Variable {inp[1]} already exists. Rename variable'
-                                    ' or delete copy of variable.')
-
-                # make dict with given var name as key and meta data from model_inputs
-                inp_dict = {inp[1]: meta for _,meta in model_inputs.items()
-                            if meta['prom_name'] == inp[0]}
-
-                # check if dict is empty (no inputs added)
-                if len(inp_dict) == 0:
-                    raise Exception(f'Promoted name {inp[0]} does not'
-                                    ' exist in model.')
-
-                # update options inputs dict with new input dict
-                self.options['inputs'].update(inp_dict)
-
-            elif isinstance(inp, str):
-                # check if variable already exists in self.options['inputs']
-                # i.e. no repeated input variable names
-                if inp in self.options['inputs']:
-                    raise Exception(f'Variable {inp} already exists. Rename variable'
-                                    ' or delete copy of variable.')
-
-                # make dict with given var name as key and meta data from model_inputs
-                inp_dict = {inp: meta for _,meta in model_inputs.items()
-                            if meta['prom_name'].endswith(inp)}
-
-                # check if provided variable appears more than once in model
-                if len(inp_dict) > 1:
-                    raise Exception(f'Ambiguous variable {inp} in inputs. To'
-                                    ' specify which one is desired, use a tuple'
-                                    ' with the promoted name and variable name'
-                                    ' instead [e.g. (prom_name, var_name)].')
+        if self.first_setup:
```

Comment:
I'm not sure it's wise to skip the input/output checking after the first setup call, especially since setup won't be called that often anyway.  Are we 100% sure that the model we contain hasn't been modified in some way since the last setup call that could make the checks fail?  Users do some crazy things sometimes...